addons/website/models/website.py

```diff
@@ -798,63 +798,64 @@ def get_unique_key(self, string, template_module=False):
         return key_copy
 
     @api.model
-    def page_search_dependencies(self, page_id=False):
+    def search_url_dependencies(self, res_ids, res_model):
         """ Search dependencies just for information. It will not catch 100%
             of dependencies and False positive is more than possible
             Each module could add dependences in this dict
             :returns a dictionnary where key is the 'categorie' of object related to the given
                 view, and the value is the list of text and link to the resource using given page
         """
         dependencies = {}
-        if not page_id:
-            return dependencies
-
-        page = self.env['website.page'].browse(int(page_id))
-        website = page.website_id or self.get_current_website()
-        url = page.url
-
-        # search for website_page with link
-        website_page_search_dom = [('view_id.arch_db', 'ilike', url)] + website.website_domain()
-        pages = self.env['website.page'].search(website_page_search_dom)
-        page_key = _('Page')
-        if len(pages) > 1:
-            page_key = _('Pages')
-        page_view_ids = []
-        for page in pages:
-            dependencies.setdefault(page_key, [])
-            dependencies[page_key].append({
+        current_website = self.get_current_website()
+
+        # search for views containing the url, including the ones linked to a
+        # website.page
+        domain = []
+        urls = []
+        for record in self.env[res_model].browse([int(res_id) for res_id in res_ids]):
+            website = record.website_id or current_website
```

Comment:
Can you suppose there is a website_id field? I think there is at least one model without it: the appointments I think? Would not this crash? (just tested: it does crash)